deb/make-keyring.sh

```diff
@@ -2,21 +2,10 @@
 
 set -eux
 
-BASE=$(pwd)/brave-keyring
-KEY_URL=${KEY_URL:-https://brave-browser-apt-release.s3.brave.com/brave-core.asc}
+BASE="$PWD/brave-keyring"
+KEY_URL="${KEY_URL:-https://brave-browser-apt-release.s3.brave.com/brave-core.asc}"
 
-if [ ! -f ./brave-core.asc ]; then
-  echo "No public key file (./brave-core.asc) detected in present working directory. Downloading)"
-  wget "$KEY_URL" -O brave-core.asc
-fi
+mkdir -p "${BASE:?}/etc/apt/trusted.gpg.d" "$BASE/usr/share/keyrings"
 
-# Uncomment when/if we decide to move the keyring file to /usr/share/keyrings/ as recommended in:
-# https://wiki.debian.org/DebianRepository/UseThirdParty
-
-if [ ! -d "${BASE}/etc/apt/trusted.gpg.d/" ]; then
-  mkdir -p "${BASE}/etc/apt/trusted.gpg.d/"
-fi
-
-gpg --no-default-keyring --keyring ./tmp.gpg --import brave-core.asc
-gpg --no-default-keyring --keyring ./tmp.gpg --export > "${BASE}/etc/apt/trusted.gpg.d/brave-browser-release.gpg"
-rm -f tmp.gpg~ tmp.gpg brave-core.asc "${BASE}/etc/apt/trusted.gpg.d/brave-browser-release.gpg~"
+wget -qO- "${KEY_URL:?}"|tee "$BASE/usr/share/keyrings/brave-browser-release.asc"|\
+    gpg --dearmor >"${BASE}/etc/apt/trusted.gpg.d/brave-browser-release.gpg"
```

Comment:
We need to make sure that this file doesn't get installed on recent Debians.

If it's installed and then deleted in the `postinst` script, then it will be flagged as a missing config file by the packaging system I think.

Reply:
Also, one more thing to test (on Ubuntu 16.04) is whether the key ends up in the right format. We have had problems in the past where GnuPG 2 would use a newer keyring format that's not compatible with apt.

Reply:
> We need to make sure that this file doesn't get installed on recent Debians.

I'll look into that.

> We have had problems in the past where GnuPG 2 would use a newer keyring format that's not compatible with apt.

I used gpg and gpg2 on Ubuntu 16.04 and they produced the same output.
```console
% docker run --rm -it ubuntu:16.04 sh -c 'apt-get update -qq && apt-get install -qqy gnupg2 wget >/dev/null 2>&1 && wget -q https://brave-browser-apt-release.s3.brave.com/brave-core.asc && for g in gpg gpg2; do "$g" --dearmor <brave-core.asc|sha256sum; done'
dbe94f3beeac156f3a22245a0febc34686ac696ad5bca06e3761be1c6a92c31c  -
dbe94f3beeac156f3a22245a0febc34686ac696ad5bca06e3761be1c6a92c31c  -
%
```

Reply:
I modified the code to use `dpkg-maintscript-helper rm_conffile` instead of `rm` (and made the old keyring a conffile, which it previously wasn't), but there's a small caveat.

When I install everything the "old way" and then install the new brave-keyring package, /etc/apt/trusted.gpg.d/brave-browser-release.gpg remains in place and /etc/apt/trusted.gpg.d/brave-browser-release.gpg.dpkg-bak is created. If I then install the new brave-keyring package again, /etc/apt/trusted.gpg.d/brave-browser-release.gpg gets deleted and /etc/apt/trusted.gpg.d/brave-browser-release.gpg.dpkg-bak remains (which is harmless and by design; it will not be used, because it doesn't have the right extension).

I have no good explanation for this behavior, but it will achieve our goal over two brave-keyring upgrades. Wdyt?

Reply:
I modified the code to use `dpkg-maintscript-helper rm_conffile` instead of `rm` (and made the old keyring a conffile, which it previously wasn't), but there's a small caveat.

When I install everything the "old way" and then install the new brave-keyring package, /etc/apt/trusted.gpg.d/brave-browser-release.gpg remains in place and /etc/apt/trusted.gpg.d/brave-browser-release.gpg.dpkg-bak is created. If I then install the new brave-keyring package again, /etc/apt/trusted.gpg.d/brave-browser-release.gpg gets deleted and /etc/apt/trusted.gpg.d/brave-browser-release.gpg.dpkg-bak remains (which is harmless and by design; it will not be used, because it doesn't have the right extension).

I have no good explanation for this behavior, but it will achieve our goal over two brave-keyring upgrades. Wdyt?

Reply:
I'm wondering whether we can find a simpler approach which doesn't involve working around dpkg.

Would the following work?

0. Remove `brave-browser-release.gpg` from `conffiles`.
1. Install `brave-browser-release.gpg` into `/usr/share/brave-keyring/` in all cases.
2. In the `postinst` script, symlink `/usr/share/brave-keyring/brave-browser-release.gpg` into `/etc/apt/trusted.gpg.d/` if on an old distro. This would go after the deletion of any existing file of that name like you already have, but using just `rm` directly.
3. In the `postrm` script, delete the `/etc/apt/trusted.gpg.d/brave-browser-release.gpg` symlink if it exists.

Reply:
Yeah, that all seems fine, but I'm not sure how appropriate creating files in maintainer scripts is. I was trying to avoid it, but given that the approach with conffiles doesn't work as cleanly, as I'd hope, this seems like the way to go.

Reply:
Ok, ready for review.